test/e2e/all_in_one_test.go

```diff
@@ -935,3 +1155,48 @@ func getPreviousGitTag(path string, cur semver.Version) (semver.Version, error)
 	}
 	return tags[curIndex-1], nil
 }
+
+// getKongProxyIP takes a Service with Kong proxy ports and returns and its IP, or fails the test if it cannot
+func getKongProxyIP(ctx context.Context, t *testing.T, env environments.Environment, svc *corev1.Service) string {
+	proxyIP := ""
+	require.NotEqual(t, svc.Spec.Type, svc.Spec.ClusterIP)
```

Comment:
It's an interesting requirement. I think this is OK for now but I could see us theoretically running the test suite within the cluster in the future.